azurerm/helpers/subscription/subscription.go

```diff
@@ -0,0 +1,40 @@
+package subscription
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func SubscriptionSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"subscription_id": {
+			Type:     schema.TypeString,
+			Optional: true,
```

Comment:
Given this is only optional for the `Subscription` Data Source - I believe we don't want this set for the `Subscriptions` Data Source? We can override the schema for the Subscription data source to add this, there's examples in the AWS Provider